boards/posix/native_posix/native_rtc.c

```diff
@@ -8,10 +8,10 @@
  * API to the native_posix (Real) Time Clock
  */
 
+#include <zephyr.h>
```

Comment:
Why are you doing this? 
if it is just for posix_trace.h then better to include that alone (even if it is with the whole path)

(I tried quite hard to keep a separation as clean as possible between the files that interact with the OS & HW models from the rest of Zephyr, to ensure they can be built as separately as possible (to be able to support the Zephyr POSIX shim/API) between other reasons)

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/20119#discussion_r339337396
https://github.com/zephyrproject-rtos/zephyr/pull/20119#discussion_r339472397

I was trying out the deprecation strategy as described above (i.e. make inclusion of posix_trace.h implicit in an essential Zephyr header, so users do not need to be concerned about it; the assumption is that all Zephyr applications are going to include kernel.h aka. zephyr.h).

If you do not like this idea, I will change it back to `#include <arch/posix/posix_trace.h>`.

Reply:
> (I tried quite hard to keep a separation as clean as possible between the files that interact with the underlying OS (and the HW models) from the rest of Zephyr, to ensure they can be built as separately as possible (to be able to support the Zephyr POSIX shim/API) between other reasons)

Maybe I do not understand what you are trying to achieve with the "posix arch" of Zephyr to the full extent, but as far as I am concerned, these files are specific to Zephyr only and simply will not run (and need to run) on any other platforms.

Reply:
> If you do not like this idea, I will change it back to #include <arch/posix/posix_trace.h>.

Wherever you added such a top level header like `zephyr.h` or `kernel.h` for the sole purpose of inheriting `arch/posix/posix_trace.h`, please do.

Reply:
> Maybe I do not understand what you are trying to achieve with the "posix arch" of Zephyr to the full extent, but as far as I am concerned, these files are specific to Zephyr only and simply will not run (and need to run) on any other platforms.

The POSIX arch can be a bit confusing in this regard: It is not a real arch but effectively a "cheat" layer which fools the Zephyr kernel and app to make it believe it runs normally on top of a CPU,  while in reality it is running on top of the Linux kernel ([see here](https://docs.zephyrproject.org/latest/boards/posix/native_posix/doc/index.html)).
The arch achieves that by mapping Zephyr threads to Linux/POSIX pthreads, and emulating a "single core" processor by ensuring only one of those threads (or an interrupt) runs at any given time, and that the switch between them happens as it would in a real CPU and when the kernel expects.
Where the zephyr kernel, app code and HW models are compiled into a normal Linux process using the underlying host toolchain and C library.
That imposes some limitations about how to do things, therefore my concerns about keeping the parts of the POSIX arch/socs/boards as decoupled as possible from the rest of Zephyr source, so that they can be easily compiled with different options, exposed to different include paths, into separate libraries which may require their own post-processing or tricks to be able to make it all work. 
The toughest limitations are set when we attempt to provide also a POSIX compliant API on top of Zephyr (even that this option is as of today disabled, it was possible for a few months to run the POSIX samples on top of native_posix).
Today some of the initial principles I followed are not necessary, but I still keep them in mind to ensure that if someday we need them again it would be easy enough to untangle things and apply a few more tricks.

Reply:
I thought you agreed to not add this top level zephyr.h here..(?)

Reply:
> > If you do not like this idea, I will change it back to #include <arch/posix/posix_trace.h>.
>
> Wherever you added such a top level header like zephyr.h or kernel.h for the sole purpose of inheriting arch/posix/posix_trace.h, please do.

It looks like I may have misunderstood what you meant by "please do" without reading the quoted text above.